lib/buildFactory.js

```diff
@@ -3,8 +3,65 @@
 const BaseFactory = require('./baseFactory');
 const Build = require('./build');
 const imageParser = require('docker-parse-image');
+const buildClusterAnnotation = 'screwdriver.cd/buildCluster';
+const hoek = require('hoek');
 let instance;
 
+/**
+ * Get build cluster name based on annotations
+ * @method getBuildClusterName
+ * @param  {Object}         config
+ * @param  {Object}         config.annotations              Annotations for the job
+ * @param  {PipelineModel}  config.pipeline                 Pipeline model
+ * @return {String}        Build cluster name
+ */
+async function getBuildClusterName({ annotations, pipeline }) {
+    const buildClusterName = annotations[buildClusterAnnotation] || '';
+    // Lazy load factory dependency to prevent circular dependency issues
+    // https://nodejs.org/api/modules.html#modules_cycles
+    // eslint-disable-next-line global-require
+    const BuildClusterFactory = require('./buildClusterFactory');
+    const buildClusterFactory = BuildClusterFactory.getInstance();
+
+    if (!buildClusterName) {
+        return buildClusterFactory.list({
+            managedByScrewdriver: true,
+            isActive: true
+        }).then((buildClusters) => {
+            if (buildClusters.length === 0) {
+                return '';
+            }
+
+            // Pick a random screwdriver build cluster
+            const randomCluster = buildClusters[Math.floor(Math.random() * buildClusters.length)];
```

Comment:
this will work with homogeneous build slaves in all clusters, if come clusters are less powerful then we will have to look at adding weight-age to various clusters.

Something to keep in mind when we bring up our own multiple clusters.

Reply:
yeah, that can be an improvement. should comment and add it to the design doc. current behavior is based off that

Reply:
https://github.com/screwdriver-cd/screwdriver/pull/1357/files#diff-a804af46780349d212d25ae24ce01ab0R142